plugins/3MFWriter/ThreeMFWriter.py

```diff
@@ -111,16 +112,21 @@ def _convertUMNodeToSavitarNode(um_node,
         savitar_node = Savitar.SceneNode()
         savitar_node.setName(um_node.getName())
 
-        node_matrix = Matrix()
+        node_matrix = um_node.getLocalTransformation()
         mesh_data = um_node.getMeshData()
-        # compensate for original center position, if object(s) is/are not around its zero position
-        if mesh_data is not None:
-            extents = mesh_data.getExtents()
-            if extents is not None:
-                # We use a different coordinate space while writing, so flip Z and Y
-                center_vector = Vector(extents.center.x, extents.center.z, extents.center.y)
-                node_matrix.setByTranslation(center_vector)
-        node_matrix.multiply(um_node.getLocalTransformation())
+
+        if center_mesh:
+            node_matrix = Matrix()
+            # compensate for original center position, if object(s) is/are not around its zero position
+            if mesh_data is not None:
+                extents = mesh_data.getExtents()
+                if extents is not None:
+                    # We use a different coordinate space while writing, so flip Z and Y
+                    center_vector = Vector(extents.center.x, extents.center.y, extents.center.z)
+                    node_matrix.setByTranslation(center_vector)
+            node_matrix.multiply(um_node.getLocalTransformation())
+        else:
+            node_matrix = um_node.getLocalTransformation()
```

Comment:
Do we. need this else? `node_matrix` is already initialised to `um_node.getLocalTransformation()`

Reply:
I ended up removing the first declaration, which removes a useless call in case `center_mesh` is true, and I think it makes the code a bit more readable.